examples/block_writer/main.go

```diff
@@ -0,0 +1,212 @@
+
+// numBlocks keeps a global count of successfully written blocks.
+var numBlocks uint64
+
+// A blockWriter writes blocks of random data into cockroach in an infinite
+// loop.
+type blockWriter struct {
+	id   string
+	db   *sql.DB
+	rand *rand.Rand
+}
+
+func newBlockWriter(db *sql.DB) blockWriter {
+	source := rand.NewSource(int64(time.Now().UnixNano()))
+	return blockWriter{
+		db:   db,
+		id:   uuid.NewUUID4().String(),
+		rand: rand.New(source),
+	}
+}
+
+// run is an infinite loop in which the blockWriter continuously attempts to
+// write blocks of random data into a table in cockroach DB.
+func (bw blockWriter) run() {
+	for {
+		blockID := bw.rand.Int63()
+		blockData := bw.randomBlock()
+		if _, err := bw.db.Exec(insertBlockStmt, bw.id, blockID, blockData); err != nil {
+			if log.V(1) {
+				log.Error(err)
+			}
+			continue
+		}
+		atomic.AddUint64(&numBlocks, 1)
+	}
+}
+
+// randomBlock generates a slice of randomized bytes. Random data is preferred
+// to prevent compression in storage.
+func (bw blockWriter) randomBlock() []byte {
+	blockSize := bw.rand.Intn(*maxBlockSizeBytes-*minBlockSizeBytes) + *minBlockSizeBytes
+	blockData := make([]byte, blockSize)
+	for i := range blockData {
+		blockData[i] = byte(bw.rand.Int() & 0xff)
+	}
+	return blockData
+}
+
+// parseFlags ensures that values taken from command line flags are valid.
+func parseFlags() error {
+	flag.Parse()
+	if *concurrency < 1 {
+		return util.Errorf("Value of 'concurrency' flag (%d) must be greater than or equal to 1", *concurrency)
+	}
+	if max, min := *maxBlockSizeBytes, *minBlockSizeBytes; max < min {
+		return util.Errorf("Value of 'max-block-bytes' (%d) must be greater than or equal to value of 'min-block-bytes' (%d)", max, min)
+	}
+	return nil
+}
+
+// maybeStartLocalServer will start a temporary local server if no database URL has been
+// provided.
+func maybeStartLocalServer() *server.TestServer {
+	if *dbURL != "" {
+		return nil
+	}
+	// Start a local test server.
+	security.SetReadFileFn(securitytest.Asset)
+	serv := server.StartTestServer(nil)
+	*dbURL = "https://root@" + serv.ServingAddr() + "?certs=test_certs"
+	return serv
+}
+
+// setupDatabase performs initial setup for the example, creating a database and
+// with a single table. If the desired table already exists on the cluster, the
+// existing table will be dropped.
+func setupDatabase() (*sql.DB, error) {
+	parsedURL, err := url.Parse(*dbURL)
+	if err != nil {
+		return nil, err
+	}
+
+	// Remove database from parsedUrl
+	q := parsedURL.Query()
+	q.Del("database")
+	parsedURL.RawQuery = q.Encode()
+
+	// Open connection to server and create a database.
+	db, err := sql.Open("cockroach", parsedURL.String())
+	if err != nil {
+		return nil, util.Errorf(err.Error())
+	}
+	if _, err := db.Exec("CREATE DATABASE IF NOT EXISTS datablocks"); err != nil {
+		return nil, util.Errorf(err.Error())
```

Comment:
Oh, yeah, it doesn't give the line number where the command was called so it loses information vs log.Fatal